src/github.ts

```diff
@@ -799,3 +799,34 @@ async function listIssueCommentReactionsPage(
     }
   }
 }
+
+let lastGitHubCallUnixMillis = 0;
+
+function registerGitHubCall() {
+  throttleGitHub();
+  gitHubCallsCounter++;
+}
+
+// should prevent throttling by GitHub
+async function throttleGitHub() {
+  // to be on a safe side target 0.7 RPS (it's 5000 requests per hour quota. Do preflight requests
+  // count against the quota?):
+  const millisSinceLastGitHubCall = Date.now() - lastGitHubCallUnixMillis;
+  if (millisSinceLastGitHubCall < 1500) {
+    const waitMs = 1500 - millisSinceLastGitHubCall;
+    // Reserve a spot just in case there are some parallel execution threads:
+    lastGitHubCallUnixMillis += 1500;
+    delay(waitMs);
+  } else {
+    lastGitHubCallUnixMillis = Date.now();
+  }
+}
+
+function delay(delayInMs: number) {
+  const startMillis = Date.now();
+  let i = 0;
+  // #NOT_MATURE: busy waiting, sorry
+  while (Date.now() - startMillis < delayInMs) {
```

Comment:
@Onexx any idea why it doesn't work when I try to replace busy waiting with promise-based waiting like this https://geshan.com.np/blog/2022/08/javascript-wait-1-second/#sleep-with-promise ?